library/ssl_tls13_client.c

```diff
@@ -318,6 +318,7 @@ static int ssl_tls13_write_key_share_ext(mbedtls_ssl_context *ssl,
             ssl, group_id, p, end, &key_exchange_len);
         p += key_exchange_len;
         if (ret != 0) {
+            MBEDTLS_SSL_DEBUG_MSG(1, ("client hello: failed generating xxdh key exchange"));
```

Comment:
It's usually useful to log the error code too, we have a macro for that. Also, when debugging the library, it's convenient to have the function name.
```suggestion
            MBEDTLS_SSL_DEBUG_RET(1, "mbedtls_ssl_tls13_generate_and_write_xxdh_key_exchange", ret);
```

Reply:
I agree that the MBEDTLS_SSL_DEBUG_RET should be used, but isn't it better to keep the text as is?
Seems like it gives a more literal description of the error - when someone will debug the library they will end up `Ctrl + F`ing the string anyway...

Reply:
`MBEDTLS_SSL_DEBUG_RET` would be consistent with the rest of the code base, and gives the location of the failure plus the error code. But that's no big deal.